src/caffe/layers/cudnn_conv_layer.cpp

```diff
@@ -90,6 +90,8 @@ void CuDNNConvolutionLayer<Dtype>::LayerSetUp(
   use_reshape_ = true;
   // When true, cached bottom and conv descriptors need to be set.
   initialized_cached_descs_ = false;
+  // In case of reusing it
+  WORKSPACE.release();
```

Comment:
What is the use case of this?

Here is what this line causes:
Initially, all the convolutional layers are setup. Each setup reserves `WORKSPASE` of size `INITIAL_WORKSPACE_SIZE`, and then the next setup release `WORKSPACE` and reserves it again by the same size.